### PR TITLE
[WIP] modules/SceAvPlayer: improve video player.

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- code is based on @bookmist  works, i have after reworks it, with remove and change something get after better result and all game works on video (exept resogun arg), but quitte instable, particularly in skip video, crash or random Unicorn error

- PR is for test some game with this issue on video no start with using avplayer module
 like on seine Gate 0, Chaos child serie, to love ru, akiba, etc...

- code here is currently lot instable, and i have two gma ewith strnage issue on video intro, it get black screen with sound continue. no rleate this code, is like this two game request any other thinks for continue keep show video frame.

# Result:
- Akiba intro story
image

- To love ru battle of ectazy (intro video)
image

- steine gate 0, intro on new game
image

- chaos child noah
image

- Yahari Game Demo Ore no Seishun Love-Kome wa Machigatteiru. Zoku
image

### CODE REVIEW

File: vita3k/modules/SceAvPlayer/SceAvPlayer.cpp [hunk 1]
```diff
@@ -219,11 +220,41 @@ static Ptr<uint8_t> get_buffer(const PlayerPtr &player, MediaType media_type,
     return buffer;
 }
 
+struct ThreadParamsEx {
+    KernelState *kernel = nullptr;
+    ThreadState *thread = nullptr;
+    SceUID thid = 0;
+    Address callback_address = 0;
+    uint wait_time = 0;
```
Comment (pent0) [inline vita3k/modules/SceAvPlayer/SceAvPlayer.cpp:228]: Reason? This seems like race condition prevention hack

Comment (Zangetsu38) [reply vita3k/modules/SceAvPlayer/SceAvPlayer.cpp:228]: like i said here, is for test only and for any other dev can look and try debug on this :) 
anhd for mysterious reason, if i no add delay that no works, but i can using 10 that works too

**Comment by sunho** [comment]
I'm actually advocate for background callback. I was thinking to make run_callback background by default. But, wait time looks extemely dubious.

**Comment by Zangetsu38** [comment]
> I'm actually advocate for background callback. I was thinking to make run_callback background by default. But, wait time looks extemely dubious.

Yes but can also works without this just in ly test is very more stable like that, just one thinks is obligatory create ither thread for execute run callbacj

**Comment by sunho** [comment]
Something I realized is that we'd like to run callbacks in order. Gotta have some queue.

**Comment by Zangetsu38** [comment]
> Something I realized is that we'd like to run callbacks in order. Gotta have some queue.

You are free for try fix video playing corectly, i have listed few game with this issue so good luck ^^

**Pull request closed without merging**